src/common/extensionMessaging.ts

```diff
@@ -28,6 +29,17 @@ export interface IExtensionMessageSender {
     sendMessage(message: ExtensionMessage, args?: any[]): Q.Promise<any>;
 }
 
+export function validateMessageWithArguments(message: MessageWithArguments): void {
```

Comment:
No, for some reason typescript wasn't complaining about me sending a "string" in place of the array. So sendMessage(Message, "myParameter");  compiled but failed horribly at runtime